specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2016-05-01/farms.json

```diff
@@ -0,0 +1,709 @@
+                        "$ref": "./Examples/Farms/List.json"
+                    }
+                },
+                "tags": [
+                    "Farms"
+                ],
+                "description": "Returns a list of all storage farms.",
+                "operationId": "Farms_List",
+                "parameters": [
+                    {
+                        "$ref": "storage.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK -- The list of storage farms has been returned.",
+                        "schema": {
+                            "$ref": "#/definitions/FarmList"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Storage.Admin/farms/{farmId}/ondemandgc": {
+            "post": {
+                "x-ms-examples": {
+                    "Start garbage collection on deleted storage objects.": {
+                        "$ref": "./Examples/Farms/StartGarbageCollection.json"
+                    }
+                },
+                "tags": [
+                    "Farms"
+                ],
+                "operationId": "Farms_StartGarbageCollection",
+                "description": "Start garbage collection on deleted storage objects.",
+                "parameters": [
+                    {
+                        "$ref": "storage.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/FarmIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK -- Garbage collection has completed."
+                    },
+                    "202": {
+                        "description": "ACCEPTED -- Garbage collection has started."
+                    }
+                },
+                "x-ms-long-running-operation": true
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Storage.Admin/farms/{farmId}/operationresults/{operationId}": {

```

Comment:
How do we get the `operationId` to be used in this method call?

Reply:
There is a bug to return the job id from when a user starts garbage collection.  Since this is preview we will have a change coming up with this change.